#!/bin/bash

PATCH="firefox-unity-support.patch"
URELEASE="trusty"

patch_firefox() {
UVER="${3}+build${2}"
for i in www-client/firefox/firefox-${1}.*.ebuild; do
	cat > ${PATCH} << EOF
--- a/${i}	2013-10-12 22:06:13.218057063 +0200
+++ b/${i}	2013-10-12 22:37:42.203441496 +0200
@@ -44,7 +44,8 @@
 # More URIs appended below...
 SRC_URI="\${SRC_URI}
 	http://dev.gentoo.org/~anarchy/mozilla/patchsets/\${PATCH}.tar.xz
-	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/\${PATCH}.tar.xz"
+	http://dev.gentoo.org/~nirbheek/mozilla/patchsets/\${PATCH}.tar.xz
+	https://launchpad.net/ubuntu/${URELEASE}/+source/\${PN}/\${PV}${UVER}/+files/\${PN}_\${PV}${UVER}.debian.tar.gz"
 
 ASM_DEPEND=">=dev-lang/yasm-1.1"
 
@@ -143,4 +144,5 @@
 }
 
 src_prepare() {
+	epatch "\${WORKDIR}/debian/patches/unity-menubar.patch"
 	# Apply our patches
EOF
	git apply ${PATCH}
	rm ${PATCH}
done
}

patch_firefox 24 1-0ubuntu1
patch_firefox 26 2-0ubuntu2
patch_firefox 27 1-0ubuntu1 ~b6
cat >> www-client/firefox/Manifest << EOF
DIST firefox_24.0+build1-0ubuntu1.debian.tar.gz 273652 SHA256 4e3f88622381def717b8f9d5c39aef916faadb88ced8c458b1496f187eadc390 SHA512 73dc738d2aab5f6dbc9faf60e61b0f1795db6ce773b13551207fc26a9ccd783a92cb0ca4f0c21f95e78b4ab74f9a62dc5d0497d2d92dca55f663d59e324b520d WHIRLPOOL 9f9aa0beb7fc69a97dd9381c88db5d62f9041dfed30f8470024f98a1a1d9cd0a8483e10ed018defd3c10673b84ddc55bdbe6d2e6049992264eeab818a36ad44e
DIST firefox_26.0+build2-0ubuntu2.debian.tar.gz 293562 SHA256 381b5531332eb952f334f85c5c05c5825aae84e0d4ece7af0c767c4d41d126c7 SHA512 38436d578ffb5ef83e24ccd4599e0ca4cc025b3a81f1b8aa7ae075e8b5e3daef69830dd0a6239d70e676710f245d63649907e54622672051bdd6ee62bf9f2c52 WHIRLPOOL c60b3fa9fbb78ec351719120d50df1277adc8b01e09402d1d9694b0578af2f7f8577bb07ccdece8b8ef1c4cafc9bb00802364d423d696543c5a05a12fbfeffda
DIST firefox_27.0~b6+build1-0ubuntu1.debian.tar.gz 276178 SHA256 bde286464203a6d65339c88d29c8fbde19cd53f4f201414a7e76c3f59a898aa0 SHA512 dff5dd1059a1121d6babe534df360fa2893c4baf3bf5a8a9ae59031502b9c489ed26ae3f31a8961a85f664faffb477554cf2605f9c840fc8b451638a87214e2b WHIRLPOOL a5b7f012cb70b45b0662aca5228d256b46a8a6adcf4eba94bc601b2e6b087adf001abb720352691f8ac0d0c780d473055fa8416ad8b86517cf5450902ec3d282
EOF
